scripts/books_script.php

```diff
@@ -0,0 +1,144 @@
+<?php
+
+error_reporting(E_ALL ^ E_NOTICE);
+
+// check if there is at least 1 argument
+if ($argc === 1) {
+  trigger_error('you have to give at least 1 argument');
+  exit();
+} else if ($argc > 3) {
+  trigger_error('you can give at most 2 arguments');
+  exit();
+}
+//set category
+if ($argv[1] !== '') {
+  $category = $argv[1];
+} else {
+  trigger_error('you have to give at least 1 argument');
+}
+//set author
+if (isset($argv[2])) {
+  $author = $argv[2];
+} else
+  $author = "";
+
+$startIndex = 0;
+$resultArray = [];
+
+$url = createRequestURL($author, $category, $startIndex);
+$array = get($url);
+$nrItems = $array['totalItems'];
+
+while ($startIndex < $nrItems - 41) { //< $nrItems
+  $items = $array['items'];
+  $resultArray = addToArray($resultArray, $items);
+  $startIndex += 40;
+
+  $url = createRequestURL($author, $category, $startIndex);
+  $array = get($url);
+}
+
+print_r("finalArrayLength: " . count($resultArray) . "\n");
+print_r("first Item: " . $resultArray[0]['title'] . "\n");
+print_r("url: " . $url);
+print_r("nrItems: " . $nrItems . "\n");
+
+return $resultArray;
+
+/**
+ * create url.
+ *
+ * @param string $author
+ *   Author name.
+ * @param string $category
+ *   Category name.
+ * @param number $startIndex
+ *   Start index of item to get
+ *
+ * @return string
+ *   url.
+ */
+function createRequestURL($author, $category, $startIndex)
+{
+  $url = 'https://www.googleapis.com/books/v1/volumes?q=""';
+  $url .= 'inauthor:"' . $author . '"';
+  $url .= 'subject:' . $category;
+  $url .= '&key=' . 'AIzaSyBrfRz440kh3BtxxpgfumpOY3IX7olF6xw';
+  $url .= '&maxResults=40&startIndex=' . $startIndex;
+  return $url;
+}
+
+/**
+ * Add ann array of items to result array.
+ *
+ * @param array $resultArray
+ *   Array to which items are added.
+ * @param array $items
+ *   Items to select.
+ *
+ * @return array
+ *   Array with all items.
+ */
+function addToArray($resultArray, $items)
+{
+  foreach ($items as $element) {
+    $industryId = $element['volumeInfo']['industryIdentifiers'];
```

Comment:
you'll need to verify input data better here. 
'volumeInfo' might not be set in that case you you'll have lots of warnings in a line like: 
```
"language" => $element['volumeInfo']['language'],
```
same for
```
"currency" => $element['saleInfo']['retailPrice']['currencyCode'],
```
You make the assumption that saleInfo is set, also that retailPrice is set under saleInfo

here is an example about retail price that is close to what we need:
```
      $saleInfo = !empty($bookRawData['saleInfo']) ? $bookRawData['saleInfo'] : array();
      if ($saleInfo) {
        $bookItem['price'] = !empty($saleInfo['retailPrice']) ? $saleInfo['retailPrice']['amount'] : NULL;
        $bookItem['currency'] = !empty($saleInfo['retailPrice']) ? $saleInfo['retailPrice']['currencyCode'] : NULL;
        $bookItem['buyLink'] = !empty($saleInfo['buyLink']) ? $saleInfo['buyLink'] : NULL;
      }
```
So... if a particular section is not provided, the final item will not that those keys or the keys should be NULL.